File: 2020/05/part_2.py
"""
day 5
"""
from typing import List
from part_1 import read_input, solve_one


def solve(data: List[str]):

    seat_ids = sorted([solve_one(bording_pass) for bording_pass in data])
    for idx in range(len(seat_ids))[1:-1]:
        prev = seat_ids[idx - 1]
        if prev + 2 == seat_ids[idx]:
            return prev + 1


def main():
    print(solve(read_input("sample.txt")))
    print(solve(read_input("input.txt")))


if __name__ == "__main__":
    main()

File: 2024/source/day01/part2.cpp
#include <algorithm>
#include <cctype>
#include <string>

#include <doctest/doctest.h>
#include <range/v3/algorithm/find_if.hpp>
#include <range/v3/numeric/accumulate.hpp>
#include <range/v3/view/reverse.hpp>
#include <range/v3/view/zip.hpp>
#include <wind/helpers.hpp>

namespace wind::day01::part1
{

template<typename T>
struct value_and_index
{
    size_t index;
    T value;
};

static auto parse(std::string_view file)
{
    auto parser = [](std::string_view) {};
    return wind::read_input(file, parser);
}

static auto solve(const std::vector<int32_t>& input)
{
    return ranges::accumulate(input, 0, [](auto acc, auto v) { return acc + v; });
}

TEST_SUITE("part2")
{
    TEST_CASE("sample")
    {
        auto input = parse("./data/day01/sample2.txt");
        CHECK_EQ(solve(input), 281);
    }

    TEST_CASE("input")
    {
        auto input = parse("./data/day01/input2.txt");
        CHECK_EQ(solve(input), 54265);
    }
}

}  // namespace wind::day01::part1
